### PR TITLE
feat(keyword-agent) : new-keyword-agent

### PR DESCRIPTION
**Pull request by ag4ums**

### CODE REVIEW

File: src/copyright/agent/identity.hpp [hunk 1]
```diff
@@ -1,6 +1,6 @@
 /*
  * Copyright (C) 2014, Siemens AG
```
Comment (shaheemazmalmmd) [inline src/copyright/agent/identity.hpp]: Please change the copyrights for all files.

**Comment by shaheemazmalmmd** [comment]
Please change the commit message according to commit conventions.

**Comment by shaheemazmalmmd** [comment]
Please add the below files to git ignore
 - src/copyright/VERSION-keyword
 - src/copyright/agent/keyword

File: src/copyright/agent/keyword.conf [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# properties file of the fossology KEYWORD agent
+# Copyright Siemens AG 2018 
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+#
+# Description: this file holds the regex configurations for the IP agent
```
Comment (GMishx) [inline src/copyright/agent/keyword.conf]: Please change to:
Description: This file holds the regex configurations for the **Keyword** agent

File: src/copyright/ui/keyword-hist.php [hunk 3]
```diff
@@ -0,0 +1,107 @@
+<?php
+/***********************************************************
+ * Copyright (C) 2010-2014 Hewlett-Packard Development Company, L.P.
```
Comment (shaheemazmalmmd) [inline src/copyright/ui/keyword-hist.php]: This line can be removed

File: src/lib/php/Dao/CopyrightDao.php [hunk 4]
```diff
@@ -338,7 +338,7 @@ public function getAllEntries($tableName, $uploadId, $uploadTreeTableName, $type
   public function getDecisions($tableName,$pfileId)
   {
     $statementName = __METHOD__.$tableName;
-    $sql = "SELECT * FROM $tableName where pfile_fk = $1 and is_enabled order by copyright_decision_pk desc";
+    $sql = "SELECT * FROM $tableName where pfile_fk = $1 and is_enabled order by keyword_decision_pk desc";
```
Comment (GMishx) [inline src/lib/php/Dao/CopyrightDao.php]: @ag4ums
The change is breaking Copyright view:
```
Warning: pg_prepare(): Query failed: ERROR: column "keyword_decision_pk" does not exist LINE 1: ...ision where pfile_fk = $1 and is_enabled order by keyword_de... ^ in /usr/local/share/fossology/lib/php/Db/Driver/Postgres.php on line 62

Fatal error: Uncaught Fossology\Lib\Exception: error executing: SELECT * FROM copyright_decision where pfile_fk = $1 and is_enabled order by keyword_decision_pk desc -- Fossology\Lib\Dao\CopyrightDao::getDecisionscopyright_decision
ERROR: column "keyword_decision_pk" does not exist LINE 1: ...ision where pfile_fk = $1 and is_enabled order by keyword_de... ^ in /usr/local/share/fossology/lib/php/Db/DbManager.php:141

Stack trace:
#0 /usr/local/share/fossology/lib/php/Db/ModernDbManager.php(51): Fossology\Lib\Db\DbManager->checkResult(false, 'SELECT * FROM c...')
#1 /usr/local/share/fossology/lib/php/Db/DbManager.php(184): Fossology\Lib\Db\ModernDbManager->prepare('Fossology\\Lib\\D...', 'SELECT * FROM c...')
#2 /usr/local/share/fossology/lib/php/Dao/CopyrightDao.php(344): Fossology\Lib\Db\DbManager->getRows('SELECT * FROM c...', Array, 'Fossology\\Lib\\D...')
#3 /usr/local/share/fossology/copyright/ui/Xpview.php(164): Fossology\Lib\Dao\CopyrightDao->getDecisions in /usr/local/share/fossology/lib/php/Db/DbManager.php on line 141
```

Comment (GMishx) [reply src/lib/php/Dao/CopyrightDao.php]: Created issue #1191

**Pull request merged**